src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/enchanting/AutoDisenchanter.java

```diff
@@ -128,16 +130,22 @@ private void transferEnchantments(ItemStack item, ItemStack book, Map<Enchantmen
         ItemMeta bookMeta = book.getItemMeta();
         ((Repairable) bookMeta).setRepairCost(((Repairable) itemMeta).getRepairCost());
         ((Repairable) itemMeta).setRepairCost(0);
-        item.setItemMeta(itemMeta);
         book.setItemMeta(bookMeta);
 
         EnchantmentStorageMeta meta = (EnchantmentStorageMeta) book.getItemMeta();
 
         for (Map.Entry<Enchantment, Integer> entry : enchantments.entrySet()) {
-            item.removeEnchantment(entry.getKey());
-            meta.addStoredEnchant(entry.getKey(), entry.getValue(), true);
+            boolean wasEnchantmentRemoved = itemMeta.removeEnchant(entry.getKey());
+            boolean stillHasEnchantment = itemMeta.getEnchants().containsKey(entry.getKey());
+
+            // Prevent future enchantment duplication
+            if (wasEnchantmentRemoved && !stillHasEnchantment)
+                meta.addStoredEnchant(entry.getKey(), entry.getValue(), true);
+            else //                                                                                                     Get Enchantment "Name"
+                Slimefun.logger().log(Level.SEVERE, "AutoDisenchanter has failed to remove enchantment \"{0}\"", entry.getKey().getKey().getKey());
```

Comment:
You're missing some Brackets [(Code Style Guidelines)](https://github.com/Slimefun/Slimefun4/blob/master/CONTRIBUTING.md#black_nib-code-style-guidelines)